src/middlewared/middlewared/plugins/cloud_backup/sync.py

```diff
@@ -168,3 +233,12 @@ async def abort(self, id_):
 
         await self.middleware.call("core.job_abort", cloud_backup["job"]["id"])
         return True
+
+    @private
+    async def validate_zvol(self, path):
+        dataset = zvol_path_to_name(path)
+        if not (
+            await self.middleware.call("vm.query_snapshot_begin", dataset, False) or
+            await self.middleware.call("vmware.dataset_has_vms", dataset, False)
+        ):
+            raise CallError("Backed up zvol must be used by a local or VMware VM")
```

Comment:
@themylogin sorry, I understand that PR was merged almost two years ago, but could you elaborate why zvol should be used by a local or VMware VM? In my case both TrueNAS and another VM are living inside Proxmox instance (QEMU virtualization), zvol attached as iSCSI disk VM. In my opinion taking snapshot and uploading it shouldn't have any problem.